test/test_build_caches.py

```diff
@@ -36,22 +36,26 @@ def test_build_caches():
 
         # also check topological order to prevent circular dependencies
         for dist_name, cache in caches.items():
-            pkgs = {}
-            print("Parsing manifest files for '%s'" % dist_name)
-            for pkg_name, pkg_xml in cache.release_package_xmls.items():
-                # Collect parsing warnings and fail if version convention are not respected
-                warnings = []
-                pkgs[pkg_name] = parse_package_string(pkg_xml, warnings=warnings)
-                for warning in warnings:
-                    if 'version conventions' in warning:
-                        errors.append('%s: %s' % (pkg_name, warning))
-                    else:
-                        print('%s: WARNING: %s' % (pkg_name, warning))
-            print("Order all packages in '%s' topologically" % dist_name)
-            try:
-                topological_order_packages(pkgs)
-            except RuntimeError as e:
-                errors.append('%s: %s' % (dist_name, e))
+            with Fold():
```

Comment:
Is this necessary?  The entirety of `test_build_caches()` is being done under a `with Fold()`, so this seems redundant unless I am missing something.

Reply:
It's basically a GHA limitation, it can't handle nested groups.
issue tracker: https://github.com/actions/toolkit/issues/1001

Example: https://github.com/ros/rosdistro/actions/runs/5779894957/job/15662805025?pr=38167
Block 8 should include block 9-12, and subsequent outputs.